### PR TITLE
feat: add function to stringify objects with Buffers

### PR DESCRIPTION
**Pull request by imurchie**
Add a general-purpose function to serialize objects while respecting Buffers. This is currently used in `appium-remote-debugger`, and it has a place in `appium-ios-device`

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -111,6 +111,40 @@ function safeJsonParse (obj) {
...
+ * except the `replacer` argument can only be a function.
+ *
+ * @param {object} obj - the object to be serialized
+ * @param {?function} replacer - function to transform the properties added to the
+ *                               serialized object
+ * @param {?number|string} space - used to insert white space into the output JSON
+ *                                 string for readability purposes. Defaults to 2
+ * returns {string} - the JSON object serialized as a string
+ */
+function jsonStringify (obj, replacer, space = 2) {
```
Comment (mykola-mokhnach) [inline lib/util.js]: replacer = null

File: lib/util.js [hunk 2]
```diff
@@ -111,6 +111,40 @@ function safeJsonParse (obj) {
   }
 }
 
+/*
+ * Stringifies the object passed in, converting Buffers into Strings for better
+ * display. This mimics JSON.stringify (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
+ * except the `replacer` argument can only be a function.
+ *
+ * @param {object} obj - the object to be serialized
+ * @param {?function} replacer - function to transform the properties added to the
```
Comment (mykola-mokhnach) [inline lib/util.js]: it is important to mention which args the function accepts

Comment (mykola-mokhnach) [reply lib/util.js]: and what is expected to return

Comment (imurchie) [reply lib/util.js]: I think that is best left to the MDN entry mentioned in the description.

File: lib/util.js (hunk 1 above)
Comment (imurchie) [reply lib/util.js]: That's not the default, though. The default is defined below.

**Pull request merged**